kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ByPassUserTaskServiceIntegrationTest.java

```diff
@@ -0,0 +1,147 @@
+import org.kie.server.api.model.ReleaseId;
+import org.kie.server.api.model.instance.TaskEventInstance;
+import org.kie.server.api.model.instance.TaskInstance;
+import org.kie.server.api.model.instance.TaskSummary;
+import org.kie.server.integrationtests.category.Smoke;
+import org.kie.server.integrationtests.shared.KieServerDeployer;
+import org.kie.server.integrationtests.shared.KieServerReflections;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class ByPassUserTaskServiceIntegrationTest extends JbpmKieServerBaseIntegrationTest {
+
+    private static ReleaseId releaseId = new ReleaseId("org.kie.server.testing", "definition-project",
+            "1.0.0.Final");
+
+    @BeforeClass
+    public static void buildAndDeployArtifacts() {
+
+        KieServerDeployer.buildAndDeployCommonMavenParent();
+        KieServerDeployer.buildAndDeployMavenProjectFromResource("/kjars-sources/definition-project");
+
+
+        kieContainer = KieServices.Factory.get().newKieContainer(releaseId);
+
+        createContainer(CONTAINER_ID, releaseId);
+    }
+
+    @Override
+    protected void addExtraCustomClasses(Map<String, Class<?>> extraClasses) throws Exception {
+        extraClasses.put(PERSON_CLASS_NAME, Class.forName(PERSON_CLASS_NAME, true, kieContainer.getClassLoader()));
+    }
+
+    @Test
+    @Category(Smoke.class)
+    public void testProcessWithUserTasks() throws Exception {
+        Long processInstanceId = processClient.startProcess(CONTAINER_ID, PROCESS_ID_USERTASK);
+        assertNotNull(processInstanceId);
+        assertTrue(processInstanceId.longValue() > 0);
+        try {
+            List<TaskSummary> taskList = taskClient.findTasksAssignedAsPotentialOwner(USER_YODA, 0, 10);
+            assertNotNull(taskList);
+
+            assertEquals(1, taskList.size());
+            TaskSummary taskSummary = taskList.get(0);
+            assertEquals("First task", taskSummary.getName());
+
+            // startTask and completeTask task
+            taskClient.startTask(CONTAINER_ID, taskSummary.getId(), USER_YODA);
+
+            Map<String, Object> taskOutcome = new HashMap<String, Object>();
+            taskOutcome.put("string_", "my custom data");
+            taskOutcome.put("person_", createPersonInstance(USER_MARY));
+
+            Map<String, Object> content = new HashMap<>();
+            content.put("name", "joda");
+            taskClient.saveTaskContent(CONTAINER_ID, taskSummary.getId(), "admin", content);
+            List<TaskEventInstance> events = taskClient.findTaskEvents(taskSummary.getId(), 0, 10);
+            assertEquals("admin", events.get(events.size() - 1).getUserId());
+
+            taskClient.saveTaskContent(CONTAINER_ID, taskSummary.getId(), "other", content);
+            events = taskClient.findTaskEvents(taskSummary.getId(), 0, 10);
+            assertEquals("other", events.get(events.size() - 1).getUserId());
```

Comment:
same as previous comment

Reply:
checking the last event is by the user being impersonated.